salt/utils/jinja.py

```diff
@@ -221,6 +199,27 @@ def uptodate():
         # there is no template file within searchpaths
         raise TemplateNotFound(template)
 
+    def destroy(self):
+        if self._close_file_client is False:
```

Comment:
Do we really need to do this little dance? We shouldn't be calling destroy in a multi-threaded situation.

Reply:
Also, if we truly need to do the dance then we should have some tests that illustrate why.

Reply:
The test added illustrates the dance.
If the class creates the file client, it should destroy it.
If a client is passed in, we don't destroy it.
If however we got passed a client, but the existing logic states that it's not a good client, then we create a client and also destroy it.